Implement stackable threshold for refill

VelizarBG opened the pull request:
This enables a refill to happen when an item stack's count falls within a certain threshold. It's similar to Tweakeroo's `handRestockPre` but utilizes IPN's more latency-immune refill. Fixes desyncs with Litematica's easy place mode on high-latency connections.

VelizarBG committed:
Implement stackable threshold for refill

- Fixes desyncs with Litematica's easy place mode

VelizarBG committed:
Move stackable threshold check to a more suitable place

blackd commented:
looks good to me
unfortunately currently the forge build is broken and I have no ETA for releasing this.
I'll probably need to reimplement the whole build system before that.

VelizarBG commented:
That's fine. I'm making this a draft for now until I figure out some weird interactions.

VelizarBG committed:
Fix lava removal desync

VelizarBG commented:
Mkay, hopefully this should do.

blackd commented:
OK, you'll have to explain the last commit.
The added code does almost the same thing as the removed just at different moment of the execution. What's the benefit of moving it there?

Especially considering that the decision if a stack should be moved at all was done only in the original method and now it's separated between the two methods.

VelizarBG committed:
Move stack count check back and shift click away all stackable items

- The latter change aids easy place by having the replaced stacks accumulate instead of being individually left out. This alleviates those desyncs that happen when the refill packets are on their way to the server but the stack the player is using runs out in the meantime before the server confirms receiving the refill packets.

VelizarBG commented:
Sure. So basically that commit fixes a desync I introduced earlier when bucketing lava on high-latency connections which I believe was caused by the `checkingItem.count > 1` check.
Now that you say it, I agree it kinda looks weird there so I moved it back.
In this new commit I also made it shift-click the stack if its item type is stackable. This aids easy place again because it doesn't just leave the replaced stacks all over the place because it combines them.
